### ISSUE TITLE
Problem running some gtk applications.

### ISSUE THREAD

**Issue by lazy-dolphin** [ifreund/river#11]
When trying to upload something in firefox, GtkFileChooser window does not open and firefox hangs. transmission-gtk and gtk3-demo-application do not run.

**Comment by ifreund** [comment]
Well, it turns out i haven't implemented xdg popup handling yet so that explains the issue with firefox. That should be a top priority. The `gtk3-demo-application` runs for me and I didn't find any issues with it, so not sure what's up there. `transmission-gtk` doesn't work for me either, will have to investigate

**Comment by lazy-dolphin** [comment]
I don't know if this helps or not but `gtk3-demo-application` and `transmission-gtk` output similar stuff when I launch them from the terminal.
```
** (gtk3-demo-application:985): WARNING **: 16:19:25.718: AT-SPI: Could not obtain desktop path or name


** (gtk3-demo-application:985): WARNING **: 16:19:25.739: AT-SPI: Could not obtain desktop path or name


** (gtk3-demo-application:985): WARNING **: 16:19:25.739: atk-bridge: get_device_events_reply: unknown signature

** (gtk3-demo-application:985): WARNING **: 16:19:25.739: atk-bridge: get_device_events_reply: unknown signature

** (gtk3-demo-application:985): WARNING **: 16:19:25.739: atk-bridge: GetRegisteredEvents returned message with unknown signature

** (gtk3-demo-application:985): WARNING **: 16:19:25.753: AT-SPI: Could not obtain desktop path or name


** (gtk3-demo-application:985): WARNING **: 16:19:25.753: atk-bridge: get_device_events_reply: unknown signature

** (gtk3-demo-application:985): WARNING **: 16:19:25.753: atk-bridge: get_device_events_reply: unknown signature

** (gtk3-demo-application:985): WARNING **: 16:19:25.753: atk-bridge: GetRegisteredEvents returned message with unknown signature
```
```
** (transmission-gtk:1029): WARNING **: 16:21:37.648: AT-SPI: Could not obtain desktop path or name


** (transmission-gtk:1029): WARNING **: 16:21:37.660: AT-SPI: Could not obtain desktop path or name


** (transmission-gtk:1029): WARNING **: 16:21:37.660: atk-bridge: get_device_events_reply: unknown signature

** (transmission-gtk:1029): WARNING **: 16:21:37.660: atk-bridge: get_device_events_reply: unknown signature

** (transmission-gtk:1029): WARNING **: 16:21:37.660: atk-bridge: GetRegisteredEvents returned message with unknown signature

** (transmission-gtk:1029): WARNING **: 16:21:37.675: AT-SPI: Could not obtain desktop path or name


** (transmission-gtk:1029): WARNING **: 16:21:37.675: atk-bridge: get_device_events_reply: unknown signature

** (transmission-gtk:1029): WARNING **: 16:21:37.675: atk-bridge: get_device_events_reply: unknown signature

** (transmission-gtk:1029): WARNING **: 16:21:37.675: atk-bridge: GetRegisteredEvents returned message with unknown signature
```
But launching a gtk application like geany, which works fine, outputs:
```
** (geany:1099): WARNING **: 16:22:36.895: AT-SPI: Could not obtain desktop path or name


** (geany:1099): WARNING **: 16:22:36.929: atk-bridge: get_device_events_reply: unknown signature

** (geany:1099): WARNING **: 16:22:36.929: atk-bridge: get_device_events_reply: unknown signature

** (geany:1099): WARNING **: 16:22:36.929: atk-bridge: GetRegisteredEvents returned message with unknown signature

** (geany:1099): WARNING **: 16:22:36.946: AT-SPI: Could not obtain desktop path or name


** (geany:1099): WARNING **: 16:22:36.946: AT-SPI: Could not obtain desktop path or name
```

**Comment by ifreund** [comment]
Alright, `transmission-gtk` and popups should both work now, but the firefox gtk file chooser is still broken.

**Comment by ifreund** [comment]
Well actually the transmission file chooser doesn't work either, which is great since it should be easier to debug than firefox.

**Comment by lazy-dolphin** [comment]
From my end, things are still the same, that is, both `transmission-gtk` and `gtk3-demo-application` run as a process but don't get displayed.

**Comment by lazy-dolphin** [comment]
Ok, so I tried launching both application several times and they only showed up few random times, which is very strange. I will try to test this in a different machine.

**Comment by lazy-dolphin** [comment]
On a different machine both these applications get viewed each time I launch them. `gtk3-demo-application` runs fine. However, `transmission-gtk`does not let me click on buttons, close it using Logo-Q, etc.

**Comment by ifreund** [comment]
Ok, so I've finally got a lead on this.  For some reason (imo a bug) gtk will never attach a buffer to a surface and commit if I set the state of the `xdg_toplevel` to tiled on all sides.  River currently tells all client programs that the are tiled because in my humble opinion client side decorations and shadows are dumb as rocks. I'll have to figure out how best to work around this.

**Comment by ifreund** [comment]
Actually @lazy-dolphin would you mind testing out this branch for me? I'm curious if it fixes your issues with transmission-gtk and the demo never launching as well <https://github.com/ifreund/river/tree/no-tile>.  Note that in its current state it screws up offsets so the views will be displaced to the south east.

**Comment by ifreund** [comment]
For reference, the gtk3 bug report: https://gitlab.gnome.org/GNOME/gtk/-/issues/2674

**Comment by lazy-dolphin** [comment]
Just tested it. `transmission-gtk`, `gtk3-demo-application` and even `GtkFileChooser` work perfectly in this branch in my machine.

**Comment by ifreund** [comment]
That's great news! Thanks for you help testing this (and reporting all the other bugs you find). I will put together some kind of workaround until the gtk bug gets fixed.

**Comment by ifreund** [comment]
This should be fixed by e1a14591773861c2ccbfdca0ae0e820d5b8410d4

**Comment by lazy-dolphin** [comment]
@ifreund : On running the same gtk applications after setting `rivertile left` layout which was introduced in recent commits,
- opening `firefox`'s `GtkFileChooser` crashes `river`.
- running `transmission-gtk` crashes `river`
- `gtk3-demo-application` and its `GtkFileChooser` runs fine.

**Comment by ifreund** [comment]
@lazy-dolphin should fixed in 37a98e1a7aa5be7d6c5803161482be5e9951b931
